app/models/support_interface/vendor_api_monitor.rb

```diff
@@ -1,66 +1,39 @@
 module SupportInterface
   class VendorAPIMonitor
+    def providers_with_api_usage_stats
+      Provider.select('providers.id,
+                      providers.name,
+                      last_syncs.last_sync as last_sync,
+                      last_decisions.last_decision as last_decision,
+                      errors.count as error_count,
+                      requests.count as request_count,
+                      all_time_requests.count,
+                      (CAST(errors.count AS FLOAT)/requests.count) * 100 as error_rate')
+        .joins("LEFT JOIN (#{VendorAPIRequest.successful.syncs.select('provider_id, MAX(vendor_api_requests.created_at) as last_sync').group('provider_id').to_sql}) last_syncs on last_syncs.provider_id = providers.id")
+        .joins("LEFT JOIN (#{VendorAPIRequest.successful.decisions.select('provider_id, MAX(vendor_api_requests.created_at) as last_decision').group('provider_id').to_sql}) last_decisions on last_decisions.provider_id = providers.id")
+        .joins("LEFT JOIN (#{VendorAPIRequest.errors.select('provider_id, COUNT(vendor_api_requests.id) as count').where("vendor_api_requests.created_at > current_date - interval '7 days'").group('provider_id').to_sql}) errors on errors.provider_id = providers.id")
+        .joins("LEFT JOIN (#{VendorAPIRequest.select('provider_id, COUNT(vendor_api_requests.id) as count').where("vendor_api_requests.created_at > current_date - interval '7 days'").group('provider_id').to_sql}) requests on requests.provider_id = providers.id").where(provider_type: 'university')
+        .joins("LEFT JOIN (#{VendorAPIRequest.select('provider_id, COUNT(vendor_api_requests.id) as count').group('provider_id').to_sql}) all_time_requests on all_time_requests.provider_id = providers.id").where(provider_type: 'university')
+    end
+
     def never_connected
-      @_never_connected = providers.left_outer_joins(:vendor_api_requests)
+      providers_with_api_usage_stats
         .includes(:vendor_api_tokens)
-        .distinct
-        .where(vendor_api_requests: { id: nil })
+        .where(all_time_requests: { count: nil })
     end
 
     def no_sync_in_24h
-      @_no_sync_in_24h ||= connected_providers.where.not(
-        id: VendorAPIRequest.successful.syncs.select(:provider_id).distinct.where('created_at > ?', 24.hours.ago),
-      ).map do |provider|
-        ProviderWithAPIUsageStats.new(provider)
-      end
+      providers_with_api_usage_stats.where("last_sync < now() - interval '24 hours'").order('last_sync DESC')
```

Comment:
`last_sync` and `last_decision` aren't returning the same numbers as in the previous query (testing locally), I think that's because we are getting all the successful syncs and then extracting the records with a created at less than the 24h interval/7 days interval for decisions. Previously we were selecting all connected providers which did not include this list of providers, that included the non successful API requests which we are after

Reply:
removing `successful` could resolve the issue, however we are excluding queries of providers which never actually attempted a decision/sync